### DISCUSSION TITLE
How to resolve file paths?

### DISCUSSION THREAD

**Discussion by Semro** [wxt-dev/wxt#375] [Q&A] [answered]
I have a few places in my extension with a reference to another file (script, image).

Examples:
<details>
  <summary>Creating a notification</summary>

```js
    createNotification('Interact with page', {
      type: 'basic',
      iconUrl: 'icons/icon128.png', // this
      title: chrome.i18n.getMessage('notification_interact_title'),
      message: chrome.i18n.getMessage('notification_interact_message'),
      buttons: [
        {
          title: chrome.i18n.getMessage('notification_interact_button'),
        },
      ],
    });
```
</details>

<details>
  <summary>Injecting content script in background.js</summary>

```js
    chrome.tabs.executeScript(
      tab.id,
      {
        allFrames: true,
        file: 'js/content.js', // this
        runAt: 'document_end',
      },
      () => {
        const e = chrome.runtime.lastError;
        if (e !== undefined) {
          console.warn('Injecting error: ', e.message);
        }
      },
    );
```
</details>

<details>
  <summary>Adding web_accessible_resources in manifest.json</summary>

  ```
  "web_accessible_resources": ["js/players/netflix/netflix.js"],
  ```
</details>

So, paths are just strings. Is there a solution that bundler would recognize path, include file in the build and change path respectively after build?

**Comment by aklinker1** [comment, marked as answer]
No, you'll need to write the paths as their final paths in the output directory.
